backends/p4fmt/p4fmt.cpp

```diff
@@ -24,7 +25,22 @@ std::stringstream getFormattedOutput(std::filesystem::path inputFile) {
         return formattedOutput;
     }
 
+    std::unordered_map<const Util::Comment *, bool> globalCommentsMap;
+
+    // Initialize the global comments map from the list of comments in the program.
+    if (!program->objects.empty()) {
+        const auto *firstNode = program->objects.front();
+        if (firstNode->srcInfo.isValid()) {
+            for (const auto *comment : firstNode->srcInfo.getAllFileComments()) {
```

Comment:
Will this work correctly? The name of the function suggests this will get comments from one file. So what if the first node is coming from `core.p4` and not the end-user's program?

Reply:
Resolved by fetching global comments from `parserDriver`.